src/Driver.cpp

```diff
@@ -28,10 +30,15 @@ void Driver::open(std::string const& uri)
     openURI(uri);
 }
 
-void Driver::read()
+bool Driver::read()
 {
-    readPacket(&buffer[0], buffer.size());
-    parseMessage(&buffer[4],buffer.size()-4);
+    int msg_size = readPacket(&buffer[0], buffer.size());
+    if(msg_size == KVH_1750_MSG_SIZE)
```

Comment:
>  What we have here is a method blindly parsing the given buffer and a method filling the buffer and returning it's size. If we want to enforce any contracts in such cases, there should be at least a comment explaining it, 

The contract is in the `iodrivers_base` API. Adding a comment if that is unclear to you is of course a great idea, but testing in the live codepath for a failure of extractPacket is definitely not a good practice. readPacket will only return a packet as sized by extractPacket by contract.

> The bug is that extractPacket returns the whole buffer, what I describe in the first comment of this PR.

This is not how I interpreted the PR message. This is even worse practice. You don't change a public API to check whether a regression has been fixed ...

Basically what you do in this PR is fix a bug, and then test *in the live code* whether the bug has been fixed, leaking that fact to your callers. It might not have been your intent, but you not only did create a dead path, you basically created a path that would be taken only if the PR *did not fix the bug it's attempting to fix*.

Unit tests is *definitely* the right choice here.